core/src/main/scala/com/yahoo/maha/core/FilterOperation.scala

```diff
@@ -219,6 +234,17 @@ sealed trait FilterRenderer[T, O] {
              grainOption: Option[Grain]): O
 }
 
+sealed trait FieldFilterRenderer[T, O] {
```

Comment:
You need access to the other column, maybe a better way to do this would be to modify the FilterRender trait itself.  E.g.
```
  def render(name: String,
             filter: T,
             literalMapper: LiteralMapper,
             column: Column,
             aliasToNameMapFull: Map[String, String],
             engine: Engine,
             grainOption: Option[Grain]): O
```
Then use this to get the other column:
```
val otherColumn: Column = column.columnContext.getColumnByName(aliasToNameMapFull(filter.compareTo))
```
And you should validate the other column is exists why before you get the render, ideally when creating request model.

Reply:
issue here is mostly concerned with not having a computed second alias in the case of two metrics being passed.  The name passed into this render function may be "ad_id", or it may be "SUM(IMPRESSIONS)" and I wanted it to be generic enough to handle both.  Metric handling coming soon.

Reply:
Are you saying the user query would have "SUM(IMPRESSIONS)" or that it would have the field name "Impressions" and the render would need to render it as "SUM(IMPRESSIONS)" ?

Reply:
By the time the query calls render in the generator, SUM(IMPRESSIONS) has been created by this inside the respective generator:
val exp = column match {
              case FactCol(_, dt, cc, rollup, _, annotations, _) =>
                s"""${renderRollupExpression(x.name, rollup)}"""
              case PrestoDerFactCol(_, _, dt, cc, de, annotations, rollup, _) =>
                s"""${renderRollupExpression(de.render(x.name, Map.empty), rollup)}"""
              case any =>
                throw new UnsupportedOperationException(s"Found non fact column : $any")
            }

Reply:
in that case, maybe we should pass in a map of Alias to Rendered SQL?

Reply:
I am now passing a map from alias to (name, rendered sql) so that all rendering functions can use which they need.

Reply:
This replaces original name being passed so that there will be less redundant code.